CRM/Dedupe/Merger.php

```diff
@@ -950,12 +950,10 @@ public static function merge($dupePairs = [], $cacheParams = [], $mode = 'safe',
   public static function skipMerge($mainId, $otherId, &$migrationInfo, $mode = 'safe', &$conflicts = []) {
 
     $conflicts = self::getConflicts($migrationInfo, $mainId, $otherId, $mode);
-
-    if (!empty($conflicts)) {
-      // if there are conflicts and mode is aggressive, allow hooks to decide if to skip merges
-      return (bool) $migrationInfo['skip_merge'];
-    }
-    return FALSE;
+    // A hook could have set skip_merge in order to alter merge behaviour.
+    // This is a something we might ideally deprecate since they really 'should'
+    // mess with the conflicts array instead.
+    return (bool) ($migrationData['skip_merge'] ?? !empty($conflicts));
```

Comment:
Should this be $migrationInfo instead of $migrationData? Other than that I'm probably not much help here.

Reply:
ohh that  might be  it - I hadn't gotten back to the failing test!